lib/options.nix

```diff
@@ -391,6 +391,39 @@ rec {
       );
     };
 
+  mkOptionalPackage =
```

Comment:
initial impression: this feels like an anti-pattern. Can we find a way to "use" an optional package without having to declare the option in a special way?